backend/pkg/api/handlers/handlers_test.go

```diff
@@ -0,0 +1,288 @@
+	"github.com/jmoiron/sqlx"
+	"github.com/pressly/goose/v3"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/crypto/bcrypt"
+)
+
+var ts *testServer
+var dataAccessor dataaccess.DataAccessor
+var postgres *embeddedpostgres.EmbeddedPostgres
+
+type testServer struct {
+	*httptest.Server
+}
+
+// Implement a get() method on our custom testServer type. This makes a GET
+// request to a given url path using the test server client, and returns the
+// response status code, headers and body.
+func (ts *testServer) get(t *testing.T, urlPath string) (int, string) {
+	rs, err := ts.Client().Get(ts.URL + urlPath)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rs.Body.Close()
+	body, err := io.ReadAll(rs.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	bytes.TrimSpace(body)
+	return rs.StatusCode, string(body)
+}
+
+func (ts *testServer) post(t *testing.T, urlPath string, data io.Reader) (int, string) {
+	rs, err := ts.Client().Post(ts.URL+urlPath, "application/json", data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rs.Body.Close()
+	body, err := io.ReadAll(rs.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+	bytes.TrimSpace(body)
+	return rs.StatusCode, string(body)
+}
+
+func (ts *testServer) parseErrorResonse(t *testing.T, body string) api_types.ApiErrorResponse {
+	resp := api_types.ApiErrorResponse{}
+	if err := json.Unmarshal([]byte(body), &resp); err != nil {
+		t.Fatal(err)
+	}
+	return resp
+}
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+
+	// wait till service initialization is completed (TODO: find a better way to do this)
+	// time.Sleep(30 * time.Second)
+
+	os.Exit(code)
+}
+
+func teardown() {
+	dataAccessor.Close()
+	ts.Close()
+	err := postgres.Stop()
+	if err != nil {
+		log.Error(err, "error stopping embedded postgres", 0)
+	}
+}
+
+func setup() {
+	configPath := flag.String("config", "", "Path to the config file, if empty string defaults will be used")
+
+	flag.Parse()
+
+	// terminate any currently running postgres instances
+	_ = exec.Command("pkill", "-9", "postgres").Run()
+
+	// start embedded postgres
+	postgres = embeddedpostgres.NewDatabase(embeddedpostgres.DefaultConfig().Username("postgres"))
+	err := postgres.Start()
+	if err != nil {
+		log.Fatal(err, "error starting embedded postgres", 0)
+	}
+
+	// connection the the embedded db and run migrations
+	tempDb, err := sqlx.Connect("postgres", "host=localhost port=5432 user=postgres password=postgres dbname=postgres sslmode=disable")
+	if err != nil {
+		log.Fatal(err, "error connection to test db", 0)
+	}
+
+	if err := goose.Up(tempDb.DB, "../../../pkg/commons/db/migrations/postgres"); err != nil {
+		log.Fatal(err, "error running migrations", 0)
+	}
+
+	// insert dummy user for testing (email: admin@admin, password: admin)
+	pHash, _ := bcrypt.GenerateFromPassword([]byte("admin"), 10)
+	_, err = tempDb.Exec(`
+      INSERT INTO users (password, email, register_ts, api_key, email_confirmed)
+      VALUES ($1, $2, TO_TIMESTAMP($3), $4, $5)`,
```

Comment:
Could insert this test user with some large number as user ID, which would make sure that new dashboards created for this user never interfere with real user dashboards.